RetailCoder.VBE/UI/Command/Refactorings/RefactorCommandBase.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.Vbe.Interop;
+using Rubberduck.Parsing;
+using Rubberduck.VBEditor;
+using Rubberduck.UI.Command;
+using System;
+
+namespace Rubberduck.UI.Command.Refactorings
+{
+    public abstract class RefactorCommandBase : CommandBase
+    {
+        protected readonly IRubberduckParser _parser;
+        protected readonly IActiveCodePaneEditor _editor;
+        protected readonly VBE _ide;
+
+        protected RefactorCommandBase(VBE ide, IRubberduckParser parser, IActiveCodePaneEditor editor)
+        {
+            _ide = ide;
+            _parser = parser;
+            _editor = editor;
+        }
+
+        protected void refactoring_InvalidSelection(object sender, EventArgs e)
+        {
```

Comment:
I see what you've done here, ...but a `protected` event handler feels awkward if it's not in the same class that's registering the event in question. Not a showstopper, but I'll try to think of a better way to do this.